docs/setup/installation.md

```diff
@@ -506,14 +516,11 @@ location /.well-known/matrix/client {
 }
 ```
 
-You should also ensure the `public_baseurl` option in `homeserver.yaml` is set
+You should always ensure the `public_baseurl` option in `homeserver.yaml` is set
 correctly. `public_baseurl` should be set to the URL that clients will use to
-connect to your server. This is the same URL you put for the `m.homeserver`
-`base_url` above.
-
-```yaml
-public_baseurl: "https://<matrix.example.com>"
-```
+connect to your server. If you use reverse proxy to serve client Well-Known URI,
+you should ensure the URL you put for the `public_baseurl` in `homeserver.yaml`
+and `m.homeserver` `base_url` in your reverse proxy config must be the same.
```

Comment:
I find this paragraph a bit confusing now... it *might* be clearer to do the following:

1. Add a new sub-heading (under "Client Well-Known URI") which is "Serve client well-known with Synapse", this would contain the new bits you added about this.
2. Put the reverse proxy bits into a new sub-heading "Serve client well-known with reverse proxy"

Maybe include the `public_baseurl` before either section then since it is needed for both?